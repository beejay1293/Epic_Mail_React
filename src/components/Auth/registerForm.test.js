import React from 'react';
import sinon from 'sinon';
import store from '../../store';
import { mount, shallow, render } from 'enzyme';
import RegisterForm, { Register } from './registerForm';
describe('Register form', () => {
  it('should render correctly the Register form', () => {
    const Form = shallow(<RegisterForm store={store}/>);
  
    expect(Form).toMatchSnapshot();
  });
});

describe('test register', () => {
  sinon.spy(Register.prototype, 'handleChange');
  sinon.spy(Register.prototype, 'handleSubmit');

  const props = {
    isAuthenticated: true,
    isLoading: true,
    user: {},
    errors: {},
    auth: () => {}
  }

  const enzymeWrapper = shallow(<Register {...props} />)

describe('test register methods', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = enzymeWrapper;
  })

  it('should test the button', () => {
    const button = wrapper.find('form');
    const e = { target: {}, preventDefault: () => {}}

    button.simulate('submit', e);
    expect(Register.prototype.handleSubmit.called).toBe(true)
  })

  it('should test the input field', () => {
    const input = wrapper.find('input').first();
    const e = { target: {}, preventDefault: ()  => {}}

    input.simulate('change', e);
    expect(Register.prototype.handleChange.called).toBe(true)

  })

  it('should test the input field', () => {
    const input = wrapper.find('input').at(1);
    const e = { target: {}, preventDefault: ()  => {}}

    input.simulate('change', e);
    expect(Register.prototype.handleChange.called).toBe(true)

  })

   it('should test the input field', () => {
    const input = wrapper.find('input').at(2);
    const e = { target: {}, preventDefault: ()  => {}}

    input.simulate('change', e);
    expect(Register.prototype.handleChange.called).toBe(true)

  })

  it('should test the input field', () => {
    const input = wrapper.find('input').at(3);
    const e = { target: {}, preventDefault: ()  => {}}

    input.simulate('change', e);
    expect(Register.prototype.handleChange.called).toBe(true)

  })

  it('should test the input field', () => {
    const input = wrapper.find('input').at(4);
    const e = { target: {}, preventDefault: ()  => {}}

    input.simulate('change', e);
    expect(Register.prototype.handleChange.called).toBe(true)

  })

  it('should test the input field', () => {
    const input = wrapper.find('input').at(5);
    const e = { target: {}, preventDefault: ()  => {}}

    input.simulate('change', e);
    expect(Register.prototype.handleChange.called).toBe(true)

  })
  

})


})

describe('test register methods', () => {
  const props = {
    isAuthenticated: true,
    isLoading: false,
    user: {},
    errors: {},
    auth: () => {}
  }
  const enzymeWrapper = shallow(<Register {...props} />);

  it('should test wrapper', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  })

})