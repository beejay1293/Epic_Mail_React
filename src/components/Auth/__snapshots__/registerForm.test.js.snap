// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register form should render correctly the Register form 1`] = `
<ContextProvider
  value={null}
>
  <withRouter(Register)
    auth={[Function]}
    errors={Object {}}
    isAuthenticated={false}
    isLoading={false}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    user={Object {}}
  />
</ContextProvider>
`;
